pkg/apiserver/registry/intelligence/networkpolicyrecommendation/rest.go

```diff
@@ -95,3 +89,74 @@ func (r *REST) NamespaceScoped() bool {
+	if existNPReco != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("networkPolicyRecommendation job exists, name: %s", npReco.Name))
+	}
+	job := new(crdv1alpha1.NetworkPolicyRecommendation)
+	job.Name = npReco.Name
+	job.Spec.JobType = npReco.Type
+	job.Spec.Limit = npReco.Limit
+	job.Spec.PolicyType = npReco.PolicyType
+	job.Spec.StartInterval = npReco.StartInterval
+	job.Spec.EndInterval = npReco.EndInterval
+	job.Spec.NSAllowList = npReco.NSAllowList
+	job.Spec.ExcludeLabels = npReco.ExcludeLabels
+	job.Spec.ToServices = npReco.ToServices
+	job.Spec.ExecutorInstances = npReco.ExecutorInstances
+	job.Spec.DriverCoreRequest = npReco.DriverCoreRequest
+	job.Spec.DriverMemory = npReco.DriverMemory
+	job.Spec.ExecutorCoreRequest = npReco.ExecutorCoreRequest
+	job.Spec.ExecutorMemory = npReco.ExecutorMemory
+	_, err := r.npRecommendationQuerier.CreateNetworkPolicyRecommendation(defaultNameSpace, job)
+	if err != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("error when create NetworkPolicyRecommendation CR: %v", err))
+	}
+	return &metav1.Status{Status: metav1.StatusSuccess}, nil
+}
+
+func (r *REST) Delete(_ context.Context, name string, _ rest.ValidateObjectFunc, _ *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	_, err := r.npRecommendationQuerier.GetNetworkPolicyRecommendation(defaultNameSpace, name)
+	if err != nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("networkPolicyRecommendation job doesn't exist, name: %s", name))
+	}
+	err = r.npRecommendationQuerier.DeleteNetworkPolicyRecommendation(defaultNameSpace, name)
+	if err != nil {
+		return nil, false, err
+	}
+	return &metav1.Status{Status: metav1.StatusSuccess}, false, nil
+}
+
+// copyNetworkPolicyRecommendation is used to copy NetworkPolicyRecommendation from crd to intelligence
+func (c *REST) copyNetworkPolicyRecommendation(intelli *intelligence.NetworkPolicyRecommendation, crd *crdv1alpha1.NetworkPolicyRecommendation) error {
+	intelli.Name = crd.Name
+	intelli.Type = crd.Spec.JobType
+	intelli.Limit = crd.Spec.Limit
+	intelli.PolicyType = crd.Spec.PolicyType
+	intelli.StartInterval = crd.Spec.StartInterval
+	intelli.EndInterval = crd.Spec.EndInterval
+	intelli.NSAllowList = crd.Spec.NSAllowList
+	intelli.ExcludeLabels = crd.Spec.ExcludeLabels
+	intelli.ToServices = crd.Spec.ToServices
+	intelli.ExecutorInstances = crd.Spec.ExecutorInstances
+	intelli.DriverCoreRequest = crd.Spec.DriverCoreRequest
+	intelli.DriverMemory = crd.Spec.DriverMemory
+	intelli.ExecutorCoreRequest = crd.Spec.ExecutorCoreRequest
+	intelli.ExecutorMemory = crd.Spec.ExecutorMemory
+	intelli.Status.State = crd.Status.State
+	intelli.Status.SparkApplication = crd.Status.SparkApplication
+	intelli.Status.CompletedStages = crd.Status.CompletedStages
+	intelli.Status.TotalStages = crd.Status.TotalStages
+	intelli.Status.RecommendedNetworkPolicy = crd.Status.RecommendedNP.Spec.Yamls
+	intelli.Status.ErrorCode = crd.Status.ErrorMsg
+	// todo: need to parse the error code
```

Comment:
Fixed.
I haven't come up with a good definition and use case of ErrorCode.

Reply:
I'm fine to remove ErrorCode if it does not have a use case. Would like to get @wsquan171 's thought on this.

Reply:
Originally the error code was added to facilitate API clients to programmatically react on different types of errors based on error code instead of error msg. For instance, a front end may prompt user differently if the error is by nature invalid input or cluster availability issue. error codes are in general better way to traceback issues in workflows compared to error msg.

For current CLI, since there's little internal retry, and propagating error msg back to stdout is good enough, the use of error code may not be that obvious. For long term serviceability considerations, I still think it's a good-to-have. cc @salv-orlando for more feedback on this.